docs/developer-guide/ksqldb-clients/dotnet-client.md

```diff
@@ -0,0 +1,229 @@
+  </ItemGroup>
+
+</Project>
+```
+
+Create your example app at `ExampleSolution/ExampleApp/ExampleApp.cs`:
+
+```csharp
+using Confluent.KsqlDb;
+
+Console.WriteLine("Test application!");
+
+IClient client = ClientBuilder
+    .NewBuilder()
+    .SetHost("localhost")
+    .SetPort(8088)
+    .SetUseTls(false)
+    .Build();
+
+...
+
+client.Close();
+
+```
+
+You can use the `ClientBuilder` class to connect your CSharp client to
+{{ site.ccloud }}. For more information, see
+[Connect to a {{ site.ccloud }} ksqlDB cluster](#connect-to-cloud).
+
+Receive query results one row at a time (StreamQueryAsync())<a name="stream-query"></a>
+----------------------------------------------------------------------------------
+
+The `StreamQueryAsync()` method enables client apps to receive query results one row at a time,
+asynchronously using the built in `await` mechanism.
+
+You can use this method to issue both push and pull queries, but the usage pattern is better for push queries.
+For pull queries, consider using the [`ExecuteQueryAsync()`](#execute-query)
+method instead.
+
+Query properties can be passed as an optional second argument along with a cancellation token 
+as an optional third argument.
+
+By default, push queries return only newly arriving rows. To start from the beginning of the stream or table,
+set the `auto.offset.reset` property to `earliest`.
+
+```csharp
+public interface IClient {
+
+  /// <summary>
+  /// Executes a query. Note that this is limited to pull and push queries and does not cover
+  /// other statement type commands.
+  /// 
+  /// This is appropriate for queries which may not run to immediate completion, so this works well
+  /// for both pull and push queries.
+  /// </summary>
+  /// <param name="sql">The sql query to run</param>
+  /// <param name="properties">The properties to send along with the query</param>
+  /// <param name="cancellationToken">The cancellation token for stopping the query</param>
+  /// <returns>A task for the result</returns>
+  Task<IStreamedQueryResult> StreamQueryAsync(string sql, IDictionary<string, object> properties = null,
+          CancellationToken cancellationToken = default);
+  ...
+  
+}
+```
+
+### Asynchronous Usage ###
+
+Due to C#'s built in async/await functionality, we can write async code without a lot of the
+complexity that in other clients.  All of our API methods return `Task` objects and can be awaited.
+
+To consume records one-at-a-time in a synchronous fashion, use the `poll()` method on the query result object.
+If `poll()` is called with no arguments, it blocks until a new row becomes available or the query is terminated.
+You can also pass a `Duration` argument to `poll()`, which causes `poll()` to return `null` if no new rows are received by the time the duration has elapsed.
+For more information, see the [API reference](api/io/confluent/ksql/api/client/StreamedQueryResult.html#poll(java.time.Duration)).
```

Comment:
We will need to change the API reference link to the c# api docs

Reply:
Ah, yes.  I've removed references to the API docs for now since I'm not entirely sure where they will live.  I can always readd them when they exist.